GitCommands/Settings/ConfigFileSettings.cs

```diff
@@ -70,16 +70,46 @@ public static ConfigFileSettings CreateGlobal(ConfigFileSettings? lowerPriority,
                 SettingLevel.SystemWide);
         }
 
-        public new string GetValue(string setting)
-        {
-            return GetString(setting, string.Empty);
-        }
+        public new string GetValue(string setting) => GetString(setting, string.Empty);
 
-        public IReadOnlyList<string> GetValues(string setting)
-        {
-            return SettingsCache.GetValues(setting);
+        /// <summary>
+        /// Get the config setting from git converted in an expected C# value type (bool, int, ...)
+        /// </summary>
+        /// <typeparam name="T">the expected type to convert the value to.</typeparam>
+        /// <param name="setting">the git setting key</param>
+        /// <returns>
+        /// null if the settings is not set
+        /// the value converted in the <typeparamref name="T" /> type otherwise.
+        /// </returns>
+        public T? GetValue<T>(string setting) where T : struct => ConvertValue<T>(GetValue(setting), setting);
+
+        private T? ConvertValue<T>(string value, string setting) where T : struct
+        {
+            // Handle case where setting is not set
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return null;
+            }
+
+            Type targetType = typeof(T);
+            try
+            {
+                return (T)Convert.ChangeType(value, targetType);
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"Setting '{setting}': fail to convert value '{value}' into type '{targetType}'. Error: {ex}");
+                throw new GitConfigFormatException($"Setting '{setting}': fail to convert value '{value}' into type '{targetType}'");
```

Comment:
Gentle ping

Reply:
That is not possible to add the original exception here because as strange as it is, as I said before:

>where the settings are read, in `OnShow()`, the exception is unwrapped by the `Control.cs` code to keep only the inner exception and so even if a `UserExternalOperationException`/`GitConfigFormatException` is thrown, **that's not the type of the one received in `BugReportInvoker.Report()`** (we receive the original exception 😢 ) and so the special behavior put in place to display a more user friendly popup can't work. We still can move some code to not do things in  `OnShow()` but not sure we can do everywhere like that.

So I think it's better not keep like it is.

Reply:
failed not fail

Reply:
Thank you for the explanation.
It may worthwhile adding a comment to the code noting why we don't pass the inner exception.